### ISSUE TITLE
Build Model Failure - keras._tf_keras.keras' has no attribute 'experimental

### ISSUE THREAD

**Issue by g5d6** [kahst/BirdNET-Analyzer#210]
Here's what occurs when I attempt to run the training script against the sample training data. Any idea why this is failing?  

hgray@compaq:~/BirdNET-Analyzer$ python3 train.py --i /home/hgray/Downloads/BirdNET_Custom_Classifier_Train_Data/train_data --o /home/hgray/Documents
Loading training data...
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
...Done. Loaded 50 training samples and 1 labels.
Building model...
...Done.
Training model...
Training on 40 samples, validating on 10 samples.
Traceback (most recent call last):
  File "/home/hgray/BirdNET-Analyzer/train.py", line 161, in <module>
    trainModel()
  File "/home/hgray/BirdNET-Analyzer/train.py", line 104, in trainModel
    classifier, history = model.trainLinearClassifier(
  File "/home/hgray/BirdNET-Analyzer/model.py", line 213, in trainLinearClassifier
    lr_schedule = keras.experimental.CosineDecay(learning_rate, epochs * x_train.shape[0] / batch_size)
AttributeError: module 'keras._tf_keras.keras' has no attribute 'experimental'

**Comment by Josef-Haupt** [comment]
Which tensorflow and keras versinos do you have installed?

**Comment by g5d6** [comment]
Hi Joseph-Haupt.  Here's what I have installed:

hgray@compaq:~$ pip show tensorflow
**Name: tensorflow
Version: 2.16.0**
Summary: TensorFlow is an open source machine learning framework for everyone.
Home-page: https://www.tensorflow.org/
Author: Google Inc.
Author-email: packages@tensorflow.org
License: Apache 2.0
Location: /home/hgray/.local/lib/python3.10/site-packages
Requires: absl-py, astunparse, flatbuffers, gast, google-pasta, grpcio, h5py, keras-nightly, libclang, ml-dtypes, numpy, opt-einsum, packaging, protobuf, setuptools, six, tb-nightly, tensorflow-io-gcs-filesystem, termcolor, tf-estimator-nightly, typing-extensions, wrapt
Required-by:
hgray@compaq:~$ pip show keras
**Name: keras
Version: 2.14.0**
Summary: Deep learning for humans.
Home-page: https://keras.io/
Author: Keras team
Author-email: keras-users@googlegroups.com
License: Apache 2.0
Location: /home/hgray/.local/lib/python3.10/site-packages
Requires:
Required-by:
hgray@compaq:~$

**Comment by g5d6** [comment]
Hi Joseph-Haupt.  Here's what I have installed.

hgray@compaq:~$ pip show tensorflow
Name: tensorflow
Version: 2.16.0
Summary: TensorFlow is an open source machine learning framework for everyone.
Home-page: https://www.tensorflow.org/
Author: Google Inc.
Author-email: packages@tensorflow.org
License: Apache 2.0
Location: /home/hgray/.local/lib/python3.10/site-packages
Requires: absl-py, astunparse, flatbuffers, gast, google-pasta, grpcio, h5py, keras-nightly, libclang, ml-dtypes, numpy, opt-einsum, packaging, protobuf, setuptools, six, tb-nightly, tensorflow-io-gcs-filesystem, termcolor, tf-estimator-nightly, typing-extensions, wrapt
Required-by:
hgray@compaq:~$ pip show keras
Name: keras
Version: 2.14.0
Summary: Deep learning for humans.
Home-page: https://keras.io/
Author: Keras team
Author-email: keras-users@googlegroups.com
License: Apache 2.0
Location: /home/hgray/.local/lib/python3.10/site-packages
Requires:
Required-by:
hgray@compaq:~$

**Comment by Josef-Haupt** [comment]
Do you use the tensorflow beta version? Because the latest release is 2.15.0.
Not sure what the problem is exactly, but you could try to downgrade tensorflow to 2.14.0 since "keras.experimanteal" should exist as per their [documentation](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/CosineDecay)

**Comment by g5d6** [comment]
That's a good course of action.  I ran into some issues early on, so I chose to build from source which may have sent me down a path I shouldn't have taken.  Thanks.

**Comment by g5d6** [comment]
This issue can be CLOSED.  I finally solved my issue after some additional experiments and research.  What I discovered was that the root of my problem appears to be related to the fact I was using had an Intel processor that didn't support Advanced Vector Extensions (AVX).  I built a new system using a fully compliant CPU with a fresh Ubuntu installation, and all my problems went away. Thanks for the comments as I worked through this issue.
